web/docs/auth/migrate-from-11.md

```diff
@@ -0,0 +1,318 @@
+---
+title: Migration from 0.11.X
+---
+
+Wasp made a big change in the way authentication works in version 0.12.0. This guide will help you migrate your app from 0.11.X to 0.12.X.
+
+## What Changed?
+
+### 0.11.X Auth Models
+
+In 0.11.X, authentication was based on the `User` model which the developer needed to set up properly and take care of the auth fields like `email` or `password`.
+
+```wasp title="main.wasp"
+app myApp {
+  wasp: {
+    version: "^0.11.0"
+  },
+  title: "My App",
+  auth: {
+    userEntity: User,
+    externalAuthEntity: SocialLogin,
+    methods: {
+      gitHub: {}
+    },
+    onAuthFailedRedirectTo: "/login"
+  },
+}
+
+entity User {=psl
+  id                        Int           @id @default(autoincrement())
+  username                  String        @unique
+  password                  String
+  externalAuthAssociations  SocialLogin[]
+psl=}
+
+entity SocialLogin {=psl
+  id          Int       @id @default(autoincrement())
+  provider    String
+  providerId  String
+  user        User      @relation(fields: [userId], references: [id], onDelete: Cascade)
+  userId      Int
+  createdAt   DateTime  @default(now())
+  @@unique([provider, providerId, userId])
+psl=}
+```
+
+### New Auth Models
+
+From 0.12.X onwards, authentication is based on the auth models which are automatically set up by Wasp. You don't need to take care of the auth fields anymore.
+
+The `User` model is now just a business logic model and you use it for storing the data that is relevant for your app.
+
+```wasp title="main.wasp"
+app myApp {
+  wasp: {
+    version: "^0.12.0"
+  },
+  title: "My App",
+  auth: {
+    userEntity: User,
+    methods: {
+      gitHub: {}
+    },
+    onAuthFailedRedirectTo: "/login"
+  },
+}
+
+entity User {=psl
+  id Int @id @default(autoincrement())
+psl=}
+```
+
+You can read more about the new auth system in the [Auth Entities](./entities) section.
+
+## How to Migrate?
+
+Migrating your existing app to the new auth system is a two-step process:
+1. Migrate to the new auth system
+1. Cleanup the old auth system
+
+:::caution Migrating a deployed app
+
+To migrate a deployed app, you need to go through the migration steps while deploying the app in between the steps.
+
+Read more about migrating a deployed app in the [Migrating a Deployed App](#migrating-a-deployed-app) section.
+:::
+
+### 1. Migrate to the New Auth System
+
+If you have an existing app with authentication set up, you can follow these steps to migrate to the new auth system:
+
```

Comment:
I would add step "zero" that tells them again to read the new docs about Auth.

Actually that might be a lot hm -> but I remember I read somewhere exact changes we did. Was that part of some PR you did? Could we use that same text to tell them what changed? It would be nice if they could see in one place, concisely, what changed, otherwise they have to read the whole docs which is PITA. This same text can also go into Changelog potentially.

Reply:
I've written an introduction on this page that says "we had these models before, now we don't have them anymore". I also them them "Read more here" if they are interested :)

Reply:
That is great, but there are also some other small details, like how can they obtain username now, right?  Some small things like that, would be good if they can quickly learn about that, instead of having to read the whole new docs.

Reply:
> That is great, but there are also some other small details, like how can they obtain username now, right? Some small things like that, would be good if they can quickly learn about that, instead of having to read the whole new docs.

I've added this in the later parts of the docs.

> Oh btw what about the fact that they can't anymore have multiple ways of auth per user? That is a regression -> what can they do about that if they had a user with multiple ways to log in?

I've updated the docs 👍